services/store.go

```diff
@@ -181,71 +186,46 @@ outer:
 					if !ok {
 						continue
 					}
-					xfer := decodeTransfer(height, info, evt)
-					if xfer == nil {
-						log.Warnf(
-							"No transfer detected for state %#v in transaction %s at height %d",
-							evt.States, info.TxHash.ToHexString(), height,
-						)
-						continue
-					}
-					if failed {
-						if evt.ContractAddress != ongAddr {
+					//check evm ong event log
+					isEvm, eventLog := checkEvmEventLog(evt)
+					if isEvm {
+						xfer, err := parseEvmOngTransferLog(eventLog)
+						if err != nil {
+							log.Warnf("prase evm ong err:%s,height:%d,txhash:%s", err, info.TxHash.ToHexString(), height)
 							continue
 						}
-						// NOTE(tav): We skip additional transfer events on
-						// failed transactions if we've already matched the gas
-						// fee.
-						if gasVerified {
+						balanceCal(xfer, evt, diffs, txn.Transfers)
+					} else {
+						xfer := decodeTransfer(height, info, evt)
+						if xfer == nil {
+							log.Warnf(
+								"No transfer detected for state %#v in transaction %s at height %d",
+								evt.States, info.TxHash.ToHexString(), height,
+							)
 							continue
 						}
-					}
-					if gasVerified {
-						xfer.isGas = false
-					} else if xfer.isGas {
-						if ori.Payer == xfer.from {
-							gasVerified = true
-						} else {
-							xfer.isGas = false
-						}
-					}
-					if xfer.from != nullAddr {
-						accts, ok := diffs[xfer.from]
-						if ok {
-							balance, ok := accts[evt.ContractAddress]
-							if ok {
-								accts[evt.ContractAddress] = balance.Sub(balance, xfer.amount)
-							} else {
-								accts[evt.ContractAddress] = (&big.Int{}).Neg(xfer.amount)
+						if failed {
+							if evt.ContractAddress != ongAddr {
+								continue
 							}
-						} else {
-							diffs[xfer.from] = map[common.Address]*big.Int{
-								evt.ContractAddress: (&big.Int{}).Neg(xfer.amount),
+							// NOTE(tav): We skip additional transfer events on
+							// failed transactions if we've already matched the gas
+							// fee.
+							if gasVerified {
+								continue
 							}
 						}
-					}
-					if xfer.to != nullAddr {
-						accts, ok := diffs[xfer.to]
-						if ok {
-							balance, ok := accts[evt.ContractAddress]
-							if ok {
-								accts[evt.ContractAddress] = balance.Add(balance, xfer.amount)
+						if gasVerified {
+							xfer.isGas = false
+						} else if xfer.isGas {
+							if ori.Payer == xfer.from {
+								gasVerified = true
 							} else {
-								accts[evt.ContractAddress] = xfer.amount
-							}
-						} else {
-							diffs[xfer.to] = map[common.Address]*big.Int{
-								evt.ContractAddress: xfer.amount,
+								xfer.isGas = false
 							}
 						}
+						balanceCal(xfer, evt, diffs, txn.Transfers)
```

Comment:
This should get assigned back to `txn.Transfers`.

Reply:
hi @tav  I has fix some problem,please review again